docs/reference/troubleshooting/troubleshooting-unbalanced-cluster.asciidoc

```diff
@@ -0,0 +1,70 @@
+[[troubleshooting-unbalanced-cluster]]
+== Troubleshooting an unbalanced cluster
+
+Since 8.6 Elasticsearch is balancing shards across data tier to achieve a good compromise between
```

Comment:
I'd avoid specifying a version here, it's not very future-proof. The docs are specific to a (minor) version, we can just describe things as they are today.

Reply:
I don't think the fact that this behaviour changed is relevant to the troubleshooting process, indeed I worry that it could cause more confusion than it solves to mention it. Let's just stick with documenting the present version, it's always ok to say that newer versions have better troubleshooting support.

Reply:
I actually recommended (I think?) mentioning 8.6 as a boundary point for behavior. There was a really big change in 8.6 for balancing decisions, which causes a lot of shard movement on upgrade, so it would be good to reassure users that what they are seeing is expected and not a cause for alarm in and of itself.

Reply:
> which causes a lot of shard movement on upgrade

I do not think this is accurate. The balanced or nearly balanced clusters are not going to have any shard movements when changing/upgrading allocator. The unbalanced ones likely did had a significant amount of shard movement before upgrade, though it was harder to notice that since there were no way to estimate the remaining moves.

Reply:
How about mentioning this in the 8.6 release notes?

Reply:
> I do not think this is accurate. The balanced or nearly balanced clusters are not going to have any shard movements when changing/upgrading allocator.

I think Dianna is right here, the weight function changed in 8.6 to include more variables than just shard count, so a cluster which is balanced according to the <8.6 balancer might indeed not be balanced when upgraded to ≥8.6.

Reply:
True, however it does not necessarily translates to a lot of movements. I think it is more accurate to represent it as it _might_ cause movements and the _count_ depends on many things.